lib/api/src/ticketAPI.cpp

```diff
@@ -0,0 +1,680 @@
+    {
+        const int ticket_len = 15;
+        // random_bytes must be (unsigned char[]) to guarantee that following
+        // modulo result is positive (i.e. in [0, 61])
+        unsigned char random_bytes[ticket_len];
+        irods::getRandomBytes(random_bytes, ticket_len);
+
+        const char characterSet[] = {'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P',
+                                     'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f',
+                                     'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v',
+                                     'w', 'x', 'y', 'z', '0', '1', '2', '3', '4', '5', '6', '7', '8', '9'};
+
+        for (int i = 0; i < ticket_len; ++i) {
+            const int ix = random_bytes[i] % sizeof(characterSet);
+            newTicket[i] = characterSet[ix];
+        }
+        newTicket[ticket_len] = '\0';
+        printf("ticket:%s\n", newTicket);
+    }
+    int ticket_manager(RxComm& conn,
+                       std::string_view command,
+                       std::string_view ticket_Identifier,
+                       std::string_view commandModifier1,
+                       std::string_view commandModifier2,
+                       std::string_view commandModifier3,
+                       std::string_view commandModifier4,
+                       bool run_as_admin)
+    {
+        ticketAdminInp_t ticketAdminInp{};
+
+        ticketAdminInp.arg1 = const_cast<char*>(command.data());
+        ticketAdminInp.arg2 = const_cast<char*>(ticket_Identifier.data());
+        ticketAdminInp.arg3 = const_cast<char*>(commandModifier1.data());
+        ticketAdminInp.arg4 = const_cast<char*>(commandModifier2.data());
+        ticketAdminInp.arg5 = const_cast<char*>(commandModifier3.data());
+        ticketAdminInp.arg6 = const_cast<char*>(commandModifier4.data());
+
+        if (run_as_admin) {
+            addKeyVal(&ticketAdminInp.condInput, ADMIN_KW, "");
+        }
+
+        const int status = rcTicketAdmin(&conn, &ticketAdminInp);
+
+        irods::at_scope_exit free_memory{[&ticketAdminInp] { clearKeyVal(&ticketAdminInp.condInput); }};
+
+        if (status < 0) {
+            // if (conn.rError) {
+            //     rError_t* Err;
+            //     rErrMsg_t* ErrMsg;
+            //     int i, len;
+            //     Err = conn.rError;
+            //     len = Err->len;
+            //     for (i = 0; i < len; i++) {
+            //         ErrMsg = Err->errMsg[i];
+            //         rodsLog(LOG_ERROR, "Level %d: %s", i, ErrMsg->msg);
+            //     }
+            // }
+            char* mySubName = NULL;
+            const char* myName = rodsErrorName(status, &mySubName);
+            // rodsLog(LOG_ERROR, "rcTicketAdmin failed with error %d %s %s", status, myName, mySubName);
+            // free(mySubName);
+            THROW(status, myName);
+            return status;
```

Comment:
All of this becomes a single line.
```cpp
THROW(status, "Ticket operation failed.");
```

We don't need to resolve the error code to an error name because the error code provides enough information.